modules/contact/src/MechanicalContactConstraint.C

```diff
@@ -1787,15 +1799,15 @@ MechanicalContactConstraint::residualEnd()
 {
   if (_component == 0 && _print_contact_nodes)
   {
-    _communicator.set_union(_current_contact_state, 0);
-    if (_current_contact_state == _old_contact_state)
-      _console << "Unchanged contact state. " << _current_contact_state.size()
+    _communicator.set_union(*_current_contact_state);
+    if (*_current_contact_state == _old_contact_state)
```

Comment:
Having the _current_contact_state be a shared pointer while the old contact state is something entirely different is weird. I suppose this is due to the conditional use of a LineSearch object. Again, if we change up the design and **always** build a default one at the FEProblem level, code like this would get a lot cleaner.